pkg/controller/network/securitygroup/controller.go

```diff
@@ -152,54 +140,54 @@ func (e *external) Create(ctx context.Context, mgd resource.Managed) (managed.Ex
 	cr.Status.SetConditions(runtimev1alpha1.Creating())
 
 	// Creating the SecurityGroup itself
-	req := e.client.CreateSecurityGroupRequest(&awsec2.CreateSecurityGroupInput{
+	req := e.sg.CreateSecurityGroupRequest(&awsec2.CreateSecurityGroupInput{
 		GroupName:   aws.String(cr.Spec.GroupName),
-		VpcId:       aws.String(cr.Spec.VPCID),
+		VpcId:       cr.Spec.VPCID,
 		Description: aws.String(cr.Spec.Description),
 	})
 
 	result, err := req.Send(ctx)
 	if err != nil {
 		return managed.ExternalCreation{}, errors.Wrap(err, errCreate)
 	}
-
 	cr.UpdateExternalStatus(awsec2.SecurityGroup{GroupId: result.GroupId})
-
-	// Authorizing Ingress permissions for the SecurityGroup
-	ingressPerms := v1alpha3.BuildEC2Permissions(cr.Spec.IngressPermissions)
-	if len(ingressPerms) > 0 {
-		air := e.client.AuthorizeSecurityGroupIngressRequest(&awsec2.AuthorizeSecurityGroupIngressInput{
-			GroupId:       aws.String(cr.Status.SecurityGroupID),
-			IpPermissions: ingressPerms,
-		})
-
-		_, err = air.Send(ctx)
-		if err != nil {
-			return managed.ExternalCreation{}, errors.Wrap(err, errAuthorizeIngress)
-		}
-	}
-
-	// Authorizing Egress permissions for the SecurityGroup
-	egressPerms := v1alpha3.BuildEC2Permissions(cr.Spec.EgressPermissions)
-	if len(egressPerms) > 0 {
-		aer := e.client.AuthorizeSecurityGroupEgressRequest(&awsec2.AuthorizeSecurityGroupEgressInput{
-			GroupId:       aws.String(cr.Status.SecurityGroupID),
-			IpPermissions: egressPerms,
-		})
-
-		_, err = aer.Send(ctx)
-		if err != nil {
-			return managed.ExternalCreation{}, errors.Wrap(err, errAuthorizeEgress)
-		}
+	// We need to save status before spec update so that it's not lost.
+	if err := e.kube.Status().Update(ctx, cr); err != nil {
+		return managed.ExternalCreation{}, errors.Wrap(err, errStatusUpdate)
 	}
 
-	return managed.ExternalCreation{}, nil
+	// NOTE(muvaf): GroupID is used as external name instead of GroupName because
+	// there are cases where only GroupID is accepted as identifier.
+	meta.SetExternalName(cr, aws.StringValue(result.GroupId))
+	return managed.ExternalCreation{}, errors.Wrap(e.kube.Update(ctx, cr), errSpecUpdate)
 }
 
 func (e *external) Update(ctx context.Context, mgd resource.Managed) (managed.ExternalUpdate, error) {
+	cr, ok := mgd.(*v1alpha3.SecurityGroup)
+	if !ok {
+		return managed.ExternalUpdate{}, errors.New(errUnexpectedObject)
+	}
 	// TODO(soorena776): add more sophisticated Update logic, once we
 	// categorize immutable vs mutable fields (see #727)
 
+	if len(cr.Spec.Ingress) > 0 {
+		_, err := e.sg.AuthorizeSecurityGroupIngressRequest(&awsec2.AuthorizeSecurityGroupIngressInput{
+			GroupId:       aws.String(meta.GetExternalName(cr)),
+			IpPermissions: ec2.GenerateEC2Permissions(cr.Spec.Ingress),
+		}).Send(ctx)
+		if err != nil && !ec2.IsRuleAlreadyExistsErr(err) {
```

Comment:
Nit: You could probably use `if resource.Ignore(ec2.IsRuleAlreadyExistsErr, err) != nil { }`